File: test_project/gen_code/tbbag.cs
namespace demo
{

    public class TbBag
    {
        private readonly System.Collections.Generic.Dictionary<int, Bag> _dataMap;
        private readonly System.Collections.Generic.List<Bag> _dataList;
        
        public TbBag(System.Collections.Generic.List<Bag> dataList)
        {
            _dataList = dataList;
            _dataMap = new System.Collections.Generic.Dictionary<int, Bag>();
            foreach(var v in _dataList)
            {
                _dataMap.Add(v.Id, v);
            }
        }
        
        public System.Collections.Generic.Dictionary<int, Bag> DataMap => _dataMap;
        public System.Collections.Generic.List<Bag> DataList => _dataList;
        
        public Bag GetOrDefault(int key)
        {
            if (_dataMap.TryGetValue(key, out var v))
            {
                return v;
            }
            return null;
        }
        
        public Bag Get(int key)
        {
            if (_dataMap.TryGetValue(key, out var v))
            {
                return v;
            }
            throw new System.Collections.Generic.KeyNotFoundException();
        }
    }

}

File: test_project/gen_code_ts/tbbag.ts
export namespace demo {

    export class TbBag {
    private readonly _dataMap: Map<number, Bag>;
    private readonly _dataList: Bag[];
    
    constructor(dataList: Bag[]) {
        this._dataList = dataList;
        this._dataMap = new Map<number, Bag>();
        for (const v of this._dataList) {
            this._dataMap.set(v.Id, v);
        }
    }
    
    public get dataMap(): Map<number, Bag> {
        return this._dataMap;
    }
    
    public get dataList(): Bag[] {
        return this._dataList;
    }
    
    public get(key: number): Bag | undefined {
        return this._dataMap.get(key);
    }
}

}
